src/lambdaHandler.ts

```diff
@@ -71,35 +71,55 @@ export function startServerAndCreateLambdaHandler<
   > = options?.context ?? defaultContext;
 
   return async function (event, context) {
-    const resultMiddlewareFns: Array<LambdaResponse<RequestHandlerResult<RH>>> =
-      [];
+    const resultMiddlewareFns: Array<
+      LambdaResponse<RequestHandlerResult<RH>, TContext>
+    > = [];
     try {
       for (const middlewareFn of options?.middleware ?? []) {
-        const resultCallback = await middlewareFn(event);
-        if (resultCallback) {
-          resultMiddlewareFns.push(resultCallback);
+        const middlewareReturnValue = await middlewareFn(event);
+        // If the middleware returns an object, we assume it's a LambdaResponse
+        if (
+          typeof middlewareReturnValue === 'object' &&
+          middlewareReturnValue !== null
+        ) {
+          return middlewareReturnValue;
+        }
+        // If the middleware returns a value, we assume it's a result middleware
```

Comment:
Maybe a bit more clear?
```suggestion
        // If the middleware returns a function, we assume it's a result callback
```